### PR TITLE
feat: Azure scalable Gateway module and docs

### LINKED ISSUE #2603: Terraform module for Azure deployment of the gateway

### PR DESCRIPTION
**Pull request by jamilbk**
Resolves #2603

### CODE REVIEW

**Comment by jamilbk** [comment]
bump @bmanifold

File: terraform/examples/azure/nat-gateway/main.tf [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  #   offer     = "0001-com-ubuntu-server-jammy"
+  #   sku       = "22_04-lts"
+  #   version   = "latest"
+  # }
+
+  # Deploy a specific version of the Gateway. Generally, we recommend using the latest version.
+  # firezone_version    = "latest"
+
+  # Override the default API URL. This should almost never be needed.
+  firezone_api_url = "wss://api.firez.one"
```
Comment (bmanifold) [inline terraform/examples/azure/nat-gateway/main.tf]: Was this meant to be uncommented and set to the firez.one domain?

File: terraform/examples/azure/nat-gateway/main.tf [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+# Create a public IP for the NAT gateway
+resource "azurerm_public_ip" "firezone" {
+  name                = "firezone-pip"
+  location            = azurerm_resource_group.firezone.location
+  resource_group_name = azurerm_resource_group.firezone.name
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+
+# OPTIONAL: Create a bastion to allow SSH access to the VMs
```
Comment (bmanifold) [inline terraform/examples/azure/nat-gateway/main.tf]: Should we add a small bit along the lines of "...while setting up the Firezone Gateway..."

File: website/src/app/kb/automate/terraform/azure/readme.mdx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+After a few minutes, you should see the Firezone Gateway(s) appear in the
+Firezone admin portal. You can now configure your Resources to use the new
+Firezone Gateway(s) you just provisioned.
+
+## Upgrading
+
+To upgrade the Firezone Gateway(s) to the latest version, simply update the
+`token` and issue a `terraform apply` which will trigger a redeployment of the
+Firezone Gateway(s).
```
Comment (bmanifold) [inline website/src/app/kb/automate/terraform/azure/readme.mdx]: When you say `update the token`, do you mean generate a new token in the portal and put it in the TF scripts?

File: website/src/app/kb/use-cases/scale-vpc-access/readme.mdx [hunk 4]
```diff
@@ -31,10 +31,8 @@ balanced across multiple Gateways for high availability.
   [Deploy a Gateway](/kb/deploy/gateways) if you haven't done so yet.
 
 <Alert color="info">
-  See our [Terraform
-  examples](https://www.github.com/firezone/firezone/tree/main/terraform/examples)
-  for examples on how to automate deploying multiple Gateways to various cloud
-  providers.
+  See our [Terraform examples](/kb/automate) for examples on how to automate
+  deploying multiple Gateways to various cloud providers.
```
Comment (bmanifold) [inline website/src/app/kb/use-cases/scale-vpc-access/readme.mdx]:
```suggestion
  See our [Terraform examples](/kb/automate) to learn how to automate
  deploying multiple Gateways to various cloud providers.
```

**Review by bmanifold — COMMENTED** [review]
Overall looks good, just a couple of minor questions

File: terraform/examples/azure/nat-gateway/main.tf (hunk 1 above)
Comment (jamilbk) [reply terraform/examples/azure/nat-gateway/main.tf]: good catch! thought I got all those

File: website/src/app/kb/automate/terraform/azure/readme.mdx (hunk 3 above)
Comment (jamilbk) [reply website/src/app/kb/automate/terraform/azure/readme.mdx]: Ah, this needs to be changed, yes. It should always pull the latest version on each apply by default.

File: website/src/app/kb/use-cases/scale-vpc-access/readme.mdx (hunk 4 above)
Comment (jamilbk) [reply website/src/app/kb/use-cases/scale-vpc-access/readme.mdx]: 👍

File: website/src/app/kb/automate/terraform/azure/readme.mdx (hunk 3 above)
Comment (jamilbk) [reply website/src/app/kb/automate/terraform/azure/readme.mdx]: fixed

File: website/src/app/kb/automate/terraform/azure/readme.mdx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+After a few minutes, you should see the Firezone Gateway(s) appear in the
+Firezone admin portal. You can now configure your Resources to use the new
+Firezone Gateway(s) you just provisioned.
+
+## Upgrading
+
+To upgrade the Firezone Gateway(s) to the latest version, simply run
+`terraform apply` again. This will trigger a redeployment of the Firezone
+Gateway(s) with the latest version by default.
```
Comment (bmanifold) [inline website/src/app/kb/automate/terraform/azure/readme.mdx]: Have you tried this?  In a quick look it doesn't seem like there is anything that would be changing in the TF state, which would mean that nothing would get redeployed.  But, I may have overlooked something.

Comment (jamilbk) [reply website/src/app/kb/automate/terraform/azure/readme.mdx]: Good catch. You're right. I've updated this guide to recommend setting the Gateway version explicitly.

**Review by bmanifold — APPROVED** [review]
LGTM

**Pull request merged**